backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ResilientClusterProvider.java

```diff
@@ -59,8 +59,8 @@ private ResilientClusterProvider(ClusterConfiguration configuration) {
         LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(),
             ImmutableList.copyOf(configuration.getHosts()).toString());
 
-        Cluster cluster = ClusterFactory.create(configuration);
```

Comment:
Ok, I'm sorry if that is the issue....

Reply:
put the creating statement outside of the Callable object makes reactor retry always retry with a old cluster which is already closed from previous retry